docs/configuration_files.rst

```diff
@@ -0,0 +1,350 @@
+
+    tables:
+      cosmic_voids:
+        radii: !scipy.stats.lognorm.rvs
+          s: 200.
+          size: 10000
+        mean: !numpy.mean
+          a: $cosmic_voids.radii
+        variance: !numpy.var
+          a: $cosmic_voids.radii
+
+
+* `Multi-column assignment`: if a function returns multiple columns, you can chose to assign them to multiple columns with different names or to a muti-column object.
+
+  Imagine you want the distribution of the circular velocities of 1000 halos following a Maxwellian distribution.
+  The histogram NumPy_ returns a 2-dimensional object. Therefore, you can choose
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        hist, bin_edges: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+  or a multi-column assignment
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        histogram: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
+
+
+* `Referencing tables: table.init and table.complete dependencies`.
+
+  There are times when your function depends on tables. In these
+  cases, you need to ensure the referenced table has the necessary content and is not empty.
+
+  Example: you want to perform a very simple abundance matching, i.e. painting galaxies within your halos.
+  You can create two tables ``halos`` and ``galaxies`` storing the halo mass and galaxy luminosities.
+  Then you can stack these two tables and store it in a third table called ``matching``.
+
+  `Beware`: referencing tables is a bit different to variables or columns
+  -- where you simply tag a dollar sign.
+
+  The challenge: let ``tableA`` be a table with columns ``c1`` and ``c2``.
+  In configuration language, ``tableA`` is the name of the job.
+  *That means, when executing the configuration file, the first thing that happens is call ``tableA``, second,  call ``tableA.c1`` and third, call ``tableA.c2``.*
+  If you used the dollar sign to reference your ``tableA`` inside a function, this function might be called before the job ``tableA`` is complete, and the table will be empty.
+
+  `The solution`: to correctly reference tables, initialise your table with ``.init``, specify their dependences with the keyword ``.depends``
+  and ensure the tables are completed before calling the function with ``.complete``. Our example:
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        halo_mass: !numpy.random.uniform
+          low: 1.0e8
+          high: 1.0e14
+          size: 20
+      galaxies:
+        luminosity: !numpy.random.uniform
+          low: 0.05
+          high: 10.0
+          size: 20
+      matching:
+        .init: !astropy.table.vstack
+          tables: [ $halos, $galaxies ]
+          .depends: [ halos.complete, galaxies.complete ]
+
+
+Import objects
```

Comment:
I think how to import objects would make more sense as second bullet point under "Variables"?